include/alpaka/rand/RandStl.hpp

```diff
@@ -35,19 +56,38 @@ namespace alpaka
     namespace rand
     {
         //#############################################################################
-        //! The standard library rand implementation.
-        class RandStl
+        //! "Tiny" state mersenne twister implementation
+        class TinyMersenneTwister
```

Comment:
I do not think that you need those intermediate classes. you most probably could directly use `rand::generator::cpu::RandomDevice{}` in the unit test. This would make it more prominent that they are limited to CPU.
The rest would need a general overhaul either way but I do not think that introducing those classes brings us anything at the moment.

Reply:
Nope, did not work since we check for the base class via an enable_if currently

Reply:
We only check for `RandBase` with the `enable_if` and cast the object to `RandBase` if it is available. This is only the special case for a class that inherits from a random generator implementation (e.g. an accelerator). If there is no `RandBase`, we directly use the class itself without any casting. So at least conceptionally it should work because it is what I intended.

Reply:
Yep, I think it's broken. I tried this first and did not manage that way.

Reply:
Ah, I see your point. You are typedefing `using RandStl = TinyMersenneTwister;`. Then you need it because `RandStl` is used by some accelerators.

Reply:
But without this additional change of the default, it should work.

Reply:
I do not think that you should specialize the `createDefault`method. We should add new methods like `createTinyMersenneTwister()`. You should not change the `RandStl` defiinition. You should change the implementation of the `createDefault` specialization for `RandStl` to not use the std mersenne twister but your one.

Reply:
Maybe I miss the point, but the previous `std::mersenne_twister` did not run on CUDA as well.

Reply:
And the implemented `cuRand` mersenne twister is actually the "tiny" MT that I am implementing right now. so we are closer to being "similar" than we were before ;D

Reply:
> The direct calls you did in the unit test are not what a user should be doing.

Exactly! In the unit tests I do:
- call the default for all ACCs
- call all other RNGs where supported, just for coverage

Reply:
Yes, but this is no problem because `createDefault` simply returned something different given a CUDA accelerator.
Changing the implementation for `createDefault` on CPU is perfectly valid.

Reply:
However, to do this, you do not need those intermediate classes and you should not replace the definition of `RandStl`.

Reply:
What I want is to have 

```
auto genMersenneTwister = alpaka::rand::generator::createMersenneTwister(
            acc,
            12345u,
            6789u
        );
```

instead of 

```
auto genMersenneTwister = alpaka::rand::generator::createDefault(
            alpaka::rand::MersenneTwister{},
            12345u,
            6789u
        );
```

This `createMersenneTwister` should then be also implemented for CUDA (by simply using the cuRand version that is returned by `createDefault` or something better matching if available).

Reply:
Or to have

```
enum class Generator{
Default,
MersenneTwister,
...
};
auto genMersenneTwister = alpaka::rand::generator::create<alpaka::rand::Generator::MersenneTwister>(
            acc,
            12345u,
            6789u
        );
```

Reply:
That's ok and something for #390 

I try not to mix those aspects, since I want to backport this "fix".
 
We can address this re-structuring from `createDefault(acc`, ...)` toward `create< Generator >(acc, ...)` in a follow-up I think.